api/reporting/schemas/reporting-codegen.rb

```diff
@@ -136,68 +261,190 @@ def to_swagger_property(column)
   to_camel_case(field[:name], false)
 end
 
 # Fix up research purpose entity fields, which don't match the column names (i.e. there's no 'rp' prefix)
-def adjust_rp_col(field, table_alias)
+def adjust_col(field)
   md = field.match(/^rp_(?<root>\w+$)/)
   projection_field = to_camel_case(field, false)
   entity_property = projection_field
   if md and md['root']
     entity_property = to_camel_case(md['root'], false)
-    "#{table_alias}.#{entity_property} AS #{projection_field}"
-  else
-    "#{table_alias}.#{entity_property}"
+    "#{TABLE_INFO[:sql_alias]}.#{entity_property} AS #{projection_field}"
+  elsif TABLE_INFO[:entity_modified_columns][field]
+     "#{TABLE_INFO[:sql_alias]}.#{TABLE_INFO[:entity_modified_columns][field]}"
+   else
+    "#{TABLE_INFO[:sql_alias]}.#{entity_property}"
   end
 end
 
-  def to_query(table_name, schema)
-  table_alias = table_name[0].downcase
+def to_query()
   "@Query(\"SELECT\\n\"\n" \
-    + schema.map do |field|
-      "+ \"  #{adjust_rp_col(field[:name], table_alias)}"
+    + COLUMNS.map do |field|
+      "+ \"  #{adjust_col(field[:name])}"
     end \
     .join(",\\n\"\n") \
     + "\\n\"\n" \
-    + "+ \"FROM Db#{to_camel_case(table_name, true)} #{table_alias}\")\n" \
-    + "  List<#{projection_name(table_name)}> getReporting#{to_camel_case(table_name, true)}s();"
+    + "+ \"FROM #{TABLE_INFO[:entity_class]} #{TABLE_INFO[:sql_alias]}\")\n" \
+    + "  List<#{TABLE_INFO[:projection_interface]}> getReporting#{to_camel_case(TABLE_INFO[:name], true)}s();"
+end
+
+sql = to_query
+
+write_output(OUTPUTS[:projection_query], sql, 'Projection Query')
+
+# Unit Test Constants
+#
+BASE_TIMESTAMP = Time.new(2015, 5, 5).freeze
+TIMESTAMP_DELTA_SECONDS = 24 * 60 * 60 # .freeze # seconds in day
+
+# N.B. some Short fields are only valid up to the number of associated enum values - 1. Fixing these
+# up by hand for now.
+def to_constant_declaration(column, index)
+  value = case column[:java_type]
+          when 'String'
+            "\"foo_#{index}\""
+          when 'Integer'
+            "%d" % [index]
+          when 'Long'
+            "%dL" % [index]
+          when 'Double'
+            "%f" % [index + 0.5]
+          when 'Boolean'
+            index.even? # just flip it every time
+          when 'Timestamp'
+            # add a day times the index to base timestamp
+            timestamp = BASE_TIMESTAMP + TIMESTAMP_DELTA_SECONDS * index
+            "Timestamp.from(Instant.parse(\"#{timestamp.strftime("%Y-%m-%dT00:00:00.00Z")}\"))"
+          else
+            if column[:is_enum]
+              column[:default_enum]
+            else
+              index.to_s
+            end
+          end
+  "public static final #{column[:java_type]} #{column[:java_constant_name]} = #{value};"
+end
+
+constants = COLUMNS.enum_for(:each_with_index) \
+  .map { |col, index| to_constant_declaration(col, index) } \
+  .join("\n")
+
+write_output(OUTPUTS[:unit_test_constants], constants, 'Unit Test Constants')
```

Comment:
If I had to do this again, I'd have a folder structure like `out/$table_name/projection.java` so all the files for a particular table were together.